lib/maxwell/adapter/adapter.ex

```diff
@@ -30,9 +30,10 @@ defmodule Maxwell.Adapter do
         case res do
           %Conn{} -> res
           {:error, _reason, _conn} -> res
-          other ->
-            raise "invalid return from #{unquote(__CALLER__.module)}" <>
-                  " expected Maxwell.Conn.t or {:error, reason}, but got: #{inspect other}"
+# according to dialyzer this can never match since previous clauses completely covered the type {'error',_,#{'__struct__':='Elixir.Maxwell.Conn', 'method':=_, 'opts':=_, 'path':=_, 'query_string':=_, 'req_body':=_, 'req_headers':=_, 'state':='error', 'url':=_, _=>_}} | #{'__struct__':='Elixir.Maxwell.Conn', 'method':=_, 'opts':=_, 'path':=_, 'query_string':=_, 'req_body':='nil', 'req_headers':=_, 'resp_body':=binary(), 'resp_headers':=_, 'state':='sent', 'status':=_, 'url':=_, _=>_}
```

Comment:
@smeevil Actually this branch is intended to display a friendly error for adapter developers who mistakenly returned some thing unexpected. So IMHO it's not a good idea to remove it.

Reply:
I totally understand that, but the code cannot be reached according to Dialyzer. 
I'll change the pull request and keep it in though.

Reply:
Sorry to force you to fight against the PLT things. :P

Reply:
No problem :)